### PR TITLE
[Feature](multi-catalog) Support refresh catalog metadata

### LINKED ISSUE #11633: [Feature](multi-catalog) Support refresh catalog metadata
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-doris/issues?q=is%3Aissue) and found no similar issues.

### Description

Support refresh catalog metadata. Only external  can use it, and maunal refresh is first support 
Syntax is as follows
1.  Refresh the specified catalog manual.
```
refresh catalog catalog_name
```
2. Periodically refresh the specified catalog, interval timeunit is s.
```
create catalog catalog_name properties("auto_refresh.enable"="true", "auto_refresh.interval"="60")
```

### Use case

_No response_

### Related issues

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by stalary**
# Proposed changes

Issue Number: close https://github.com/apache/doris/issues/11633

## Problem summary

1. Support refresh catalog metadata
2. Fix a few code issues(Repeat code, warning code)

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [x] No
    - [ ] I don't know
4. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [x] No Need
5. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [x] No Need
6. Does it need to update dependencies:
    - [ ] Yes
    - [x] No
7. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [x] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: fe/fe-core/src/main/java/org/apache/doris/datasource/DataSourceMgr.java [hunk 1]
```diff
@@ -95,6 +96,20 @@ private DataSourceIf removeCatalog(long catalogId) {
...
 
+    private void unprotectedRefreshCatalog(long catalogId) {
+        DataSourceIf catalog = idToCatalog.get(catalogId);
+        if (catalog != null) {
+            String catalogName = catalog.getName();
+            if (!catalogName.equals(InternalDataSource.INTERNAL_DS_NAME)) {
+                ((ExternalDataSource) catalog).setInitialized(false);
+            }
+            catalog = nameToCatalog.get(catalogName);
+            ((ExternalDataSource) catalog).setInitialized(false);
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/datasource/DataSourceMgr.java:107]: The catalog save in `idToCatalog` and `nameToCatalog` is same object instance. So no need to set them twice.

File: fe/fe-core/src/main/java/org/apache/doris/persist/OperationType.java [hunk 2]
```diff
@@ -234,8 +237,11 @@ public class OperationType {
     public static final short OP_ALTER_DS_NAME = 314;
     public static final short OP_ALTER_DS_PROPS = 315;
     public static final short OP_ALTER_STORAGE_POLICY = 316;
+    public static final short OP_REFRESH_DS = 317;
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/persist/OperationType.java:240]:
```suggestion
    public static final short OP_REFRESH_CATALOG = 317;
```

File: fe/fe-core/src/main/java/org/apache/doris/datasource/DataSourceMgr.java (hunk 1 above)
Comment (stalary) [reply fe/fe-core/src/main/java/org/apache/doris/datasource/DataSourceMgr.java:107]: My fault. I'll fix it later.

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**